### DISCUSSION TITLE
A wrong answer from Cache record

### DISCUSSION THREAD

**Discussion by terryweijian** [zilliztech/GPTCache#385] [Q&A] [answered]
Hi 

My script contains the follow questions, and it will run several loop within one session.
 1    'what is TV ?',
 2   'can you explain what function of TV is ?',
 3    'can you tell me more about TV ?',
 4    'what is the function of money ?',

question 1~3 are all about TV, so cache answer are linked to the same answer of the first question in the first loop, but from second loop the second question will linked with the forth question of " what is the function of money". I guess there are both the key prompt word of function. Is there parameters can control the weight of vector calculation for different key words?

**The first loop:**
Question: what is TV ?
local answer: TV is a television channel that broadcasts live television.
Local Time Spent = 0.2
Cache answer: TV is a television channel that broadcasts live television.
Cache Hit Time Spent = 0.39
_______________________________________________________________
**_Question: can you explain what function of TV is ?_**
local answer: a television channel
Local Time Spent = 0.09
_Cache answer: TV is a television channel that broadcasts live television._  # **the answer is reasonable**
Cache Hit Time Spent = 0.04
_______________________________________________________________
Question: can you tell me more about TV ?
local answer: a tv show
Local Time Spent = 0.11
Cache answer: TV is a television channel that broadcasts live television.
Cache Hit Time Spent = 0.03
_______________________________________________________________
Question: what is the function of money ?
local answer: money is a currency
Local Time Spent = 0.1
Cache answer: money is a currency
Cache Hit Time Spent = 0.11
_______________________________________________________________

**Second Loop**

Question: what is TV ?
local answer: TV is a television channel that broadcasts live television.
Local Time Spent = 0.17
Cache answer: TV is a television channel that broadcasts live television.
Cache Hit Time Spent = 0.03
_______________________________________________________________
_Question: can you explain what function of TV is ?_
local answer: a television channel
Local Time Spent = 0.1
_Cache answer: money is a currency_  # **The cache answer is incorrect as it links to question 4 just because there are both have the key word of "function"** 
Cache Hit Time Spent = 0.04
_______________________________________________________________
Question: can you tell me more about TV ?
local answer: a tv show
Local Time Spent = 0.12
Cache answer: TV is a television channel that broadcasts live television.
Cache Hit Time Spent = 0.03
_______________________________________________________________
Question: what is the function of money ?
local answer: money is a currency
Local Time Spent = 0.11
Cache answer: money is a currency
Cache Hit Time Spent = 0.03
_______________________________________________________________

**Comment by SimFG** [comment, marked as answer]
@terryweijian there is no idea to control the weight of vector calculation for different key words. You can choose to skip cache searching when you think the cached answer doesn't meet the requirements, but save the llm result to the cache this time. The next time you ask the same question, you will be able to get an accurate answer.

`cache_skip` param usage
```python
openai.ChatCompletion.create(
    model="gpt-3.5-turbo",
    messages=[{"role": "user", "content": "what's github"}],
    cache_skip=True,
)
```

**Reply by terryweijian** [reply]
Not quite sure if the following step can fix the problem, I tried it and didn't find the problem in my environment again.

 use embedding_func=onnx.to_embeddings to replace my original huggingface.to_embedding

in the llm_cache.init ()

Thanks

**Reply by terryweijian** [reply]
After more testing, and confirmed that the Huggingface embedding module is the root cause, after switch to onnx embedding module, didn't find the similar issue with the English questions

Thanks
Terry
